kani-compiler/src/codegen_cprover_gotoc/codegen/contract.rs

```diff
@@ -65,11 +79,11 @@ impl<'tcx> GotocCtx<'tcx> {
         let location_of_recursion_wrapper = self.codegen_span(&span_of_recursion_wrapper);
 
         let full_name = format!(
-            "{}:{}::REENTRY",
+            "{}:{}",
             location_of_recursion_wrapper
                 .filename()
                 .expect("recursion location wrapper should have a file name"),
-            tcx.item_name(recursion_wrapper_id),
+            recursion_tracker.name(),
```

Comment:
Given that this is the second time we've had to fix the exact same issue I would add more documentation.
I would refactor this into a variable and add a comment that explains that this needs to use the same name that is emitted into the symbol table. 
At the same time I would add a comment where the name is selected for the symbol table and link it to this one (or maybe the original issue) explaining that these two names must be the same or contracts will silently fail.

Reply:
I added more documentation to highlight some of these issues.